--- templates.py.orig	2014-11-14 23:54:37.940517369 +0000
+++ templates.py	2014-11-14 23:56:15.410750577 +0000
@@ -356,6 +356,14 @@
             os.path.basename(sfn).split('.')[0],
             sfn
             )
+    # File templates need these set as __var__
+    if not '__env__' in kwargs:
+	setattr(mod, '__env__', kwargs['saltenv'])
+	builtins = ['salt', 'grains', 'pillar', 'opts']
+	for builtin in builtins:
+	    arg = '__{0}__'.format(builtin)
+	    setattr(mod, arg, kwargs[builtin])
+
     for kwarg in kwargs:
         setattr(mod, kwarg, kwargs[kwarg])
 
